File: golang/1732.go
func largestAltitude(gain []int) int {
    maxx := 0
	cur := 0
	length := len(gain)

	for i:=0; i < length; i++ {
		cur += gain[i]
		if cur > maxx {
			maxx = cur
		}
	}
	if maxx < 0 {
		maxx = 0
	}
	return maxx
}

File: rust/1732.rs
impl Solution {
    pub fn largest_altitude(gain: Vec<i32>) -> i32 {
        let mut max = 0;
        let mut cur = 0;
        for i in gain {
            cur += i;
            if cur > max {
                max = cur;
            }
        }
        max
    }
}
